### DISCUSSION TITLE
Windows / ARM64 support (specifically the `node` that's executed)

### DISCUSSION THREAD

**Discussion by SCWells72** [forcedotcom/cli#3029] [Ideas]
I just picked up a Surface Laptop 7 which runs Windows 11 on ARM64. Obviously just as with Apple's relatively recent move to their M1/2/3 chips, you get the most benefit from native apps. I've installed an ARM-native distribution of Node.js 20.17.0 on the machine, but it looks like the CLI is still running the bundled `node.exe`:

```
"C:\Users\Scott\AppData\Local\sf\client\2.58.7-ed27925\bin\..\bin\node.exe"   "C:\Users\Scott\AppData\Local\sf\client\2.58.7-ed27925\bin\..\bin\run" <args>
```

I know I can install the CLI via `npm` instead, but is there a way to have the packaged installation use the system's existing Node distribution?

Perhaps the best option would be for you to provide a packaged installer for Windows / ARM64 just as you have installers now for Windows x64 and x86 (who uses that?!) and Mac x64 and M*.

**Comment by SCWells72** [comment]
FYI, I'm also getting this failure when trying to install LWC local dev:

```
npm error command failed
npm error command C:\Windows\system32\cmd.exe /d /s /c node install.js
npm error Unsupported platform: win32 arm64 LE
```

That same plugin installs just fine when I install the Salesforce CLI via the packaged installer, so perhaps it's just an artificial check for supported platforms?

I may switch back to the packaged installer to get things installed, then see if I can hack up its front-door script to use the ARM-native build of node instead of the bundled one.

**Comment by mdonnalley** [comment]
>I know I can install the CLI via npm instead, but is there a way to have the packaged installation use the system's existing Node distribution?

Not currently (you could always swap out the included node binary for your system's but you have to do it every time you update). npm might be your best bet for now

>Perhaps the best option would be for you to provide a packaged installer for Windows / ARM64 just as you have installers now for Windows x64 and x86 (who uses that?!) and Mac x64 and M*.

Totally agree. I'll create an issue in oclif and post it here for visibility

https://github.com/oclif/oclif/issues/1557

**Comment by SCWells72** [comment]
@mdonnalley, yeah, I'm currently swapping the `node.exe` binary exactly as you suggested, and I imagine that may become enough of a nuisance that I switch to an npm-based install.

One additional question about LWC local dev...even when it's using the x64 `node.exe`, I'm getting an error when trying to start the local dev server:

```
$ sf lightning dev app
Error (1): The requested resource does not exist
```

Do you have any idea might be causing that? I'm in the process of integrating LWC local dev into IC2 and would love to use the new machine for that work, but obviously that's caused me to have to use my x64 machine instead.

I'm happy to enable any type of verbose logging that might help with troubleshooting. And I'm also happy to log that issue separately if that makes more sense (and I'm guessing it does).

**Reply by mdonnalley** [reply]
For that you'd get better support if you make an issue over here: https://github.com/salesforcecli/plugin-lightning-dev/issues

**Comment by VivekMChawla** [comment]
@SCWells72 - Thanks for bringing this to our attention. 

@mdonnalley highlighted this request for prioritization, and I agree. 

We'll make a Windows ARM64 installer available as soon as possible.

**Comment by mdonnalley** [comment]
@SCWells72 we have a arm64 installer on the nightly channel, if you're interested in giving it a try. Assuming everything's good, it'll be available on stable next Wednesday

https://developer.salesforce.com/media/salesforce-cli/sf/channels/nightly/sf-arm64.exe

**Reply by SCWells72** [reply]
Thanks, @mdonnalley. I'll be getting my own build out tomorrow and can check it out after that. I'll let you know how it goes. Thanks!

**Reply by SCWells72** [reply]
@mdonnalley, I installed the ARM64 build this morning. There was a warning about it not being signed, presumably just an aspect of it being a pre-release build, and it fails when I try to update it (assuming there's likely a new CLI build today):

```
sf update
@salesforce/cli: Updating CLI... !
    Error: HTTP 403: Invalid channel stable
```

But it does show the expected version information:

```
sf --version
@salesforce/cli/2.62.0 win32-arm64 node-v20.17.0
```

For now I think I'll go back to an npm-based install and try this again when it's available as an official/stable release next week. Thanks for getting this done so quickly!

**Reply by mdonnalley** [reply]
Awesome - thanks for testing it out! The error on `sf update` makes sense to me because we don't have a stable win32-arm64 build yet. That should be available next Wednesday

**Reply by SCWells72** [reply]
Hi, @mdonnalley. Do you know when the new packaged installer for Windows/ARM64 will be available? I'm still only seeing download links for x64 and x86 here:

https://developer.salesforce.com/tools/salesforcecli#

**Reply by mdonnalley** [reply]
@SCWells72 We're still working on the team that owns that web page. In the meantime, you can access it here: https://developer.salesforce.com/media/salesforce-cli/sf/channels/stable/sf-arm64.exe

**Reply by SCWells72** [reply]
Thanks, @mdonnalley! Installed and verified that it does in fact update in-place properly. I still have to temporarily swizzle `node.exe` to the x64 version to install `@salesforce/plugin-lightning-dev` because of the following:

```
npm error command failed
npm error command C:\Windows\system32\cmd.exe /d /s /c node install.js
npm error Unsupported platform: win32 arm64 LE
```

but once installed, that plugin works perfectly. It'd be nice if they could remove whatever check is failing there since it's a false positive.

**Reply by SCWells72** [reply]
Hmmmm...its seems that I spoke too soon. I'm actually unable to install `@salesforce/plugin-lightning-dev` on a package-based install of the CLI even swizzling both the bundled `node.exe` and the one from the local Node.js install for the x64 binary. It still fails with that same error which means to me that it's picking up `node` from somewhere else than those two binaries.

For now I've gone back to an NPM-based install because I need that plugin installed for my current work. I'm happy to help troubleshoot if someone responsible for that plugin wants to reach out.

**Comment by cristiand391** [comment]
the download page was updated to include the new win arm installers, closing this...
https://developer.salesforce.com/tools/salesforcecli#
